sycl/test-e2e/Regression/barrier_waitlist_with_interop_event.cpp

```diff
@@ -0,0 +1,44 @@
+// REQUIRES: level_zero, level_zero_dev_kit
+// RUN: %{build} %level_zero_options -o %t.out
+// RUN: %{run} %t.out
+// UNSUPPORTED: ze_debug
+
+#include <level_zero/ze_api.h>
+#include <sycl/detail/core.hpp>
+#include <sycl/ext/oneapi/backend/level_zero.hpp>
+#include <sycl/usm.hpp>
+
+// Test checks the case when an interoperability event is passed as a dependency
+// to the barrier. In such case, waiting for the event produced by barrier must
+// guarantee completion of the interoperability event.
+
+using namespace sycl;
+
+int main() {
+  sycl::queue Queue;
+  const size_t N = 1024;
+  int *Data = sycl::malloc_shared<int>(N, Queue);
```

Comment:
I think we should first check if shared allocations are available on the device. Like:

```
 if (!dev.get_info<info::device::usm_shared_allocations>())
    return 0;
```

Reply:
Indeed, fixed.